plotter/main.go

```diff
@@ -863,6 +866,30 @@ func Example_stackedBarChart() *plot.Plot {
 	return p
 }
 
+func Example_heatMap() *plot.Plot {
+	m := mat64.NewDense(3, 4, []float64{
+		1, 2, 3, 4,
+		5, 6, 7, 8,
+		9, 10, 11, 12,
+	})
+	h := plotter.NewHeatMap(m, palette.Heat(12, 1))
+
+	p, err := plot.New()
+	if err != nil {
+		panic(err)
+	}
+	p.Title.Text = "Heat map"
+
+	p.Add(h)
+
+	p.X.Padding = 0
```

Comment:
I can't say that I am terribly familiar with the context, but the insanity of matlab plots has taught me that the answer is probably functional programming via delegates.  What you want is a `func(*plot.Plot)` which deals with the formatting and can be easily chained.  The "undo" functionality that Pike uses would be a nice add.